src/graph.rs

```diff
@@ -424,6 +424,16 @@ impl Value {
         }
     }
 
+    pub fn as_bool(&self, graph: &Graph) -> Result<&bool, ExecutionError> {
```

Comment:
For `as_bool` and `as_integer`, I think it's worth having the return value be `bool` / `u32` instead of `&bool` / `&u32`.